src/helpers/logDecorator.ts

```diff
@@ -0,0 +1,39 @@
+import showTable from "../methods/showTable.js"
+import { defaultOptions } from "../types.js"
+import log from "./log.js"
+
+export default function logDecorator(simpleData: any, func: Function, ...args: any[]) {
+
+    const start = Date.now()
+
+    const parameters = args
+    let options
+
+    if (
+        args.length > 0 &&
+        typeof args[args.length - 1] === "object" &&
+        Object.keys(defaultOptions).includes(Object.keys(args[args.length - 1])[0])
+    ) {
+        options = { ...simpleData.options, ...args[args.length - 1] }
+        parameters[args.length - 1] = options
+    } else {
+        options = simpleData.options
+        parameters.push(options)
+    }
+
+    options.logs && log("\n" + func.name + "()")
+    options.logOptions && log("options:")
+    options.logOptions && log(options)
+    options.logParameters && log("parameters:")
+    options.logParameters && log(parameters.slice(0, parameters.length - 1))
+
+    const data = func(simpleData.data, ...parameters)
```

Comment:
I think that the logger should not be responsible for actually running the function. Neither should it be responsible for overwritting the data in `SimpleData `.

I would suggest something like this:
```
    addItems(...args: any[]) {
       const [params, options] = this.getParamsAndOptions(args)

        const start = Date.now()
        const data = addItems_(this, params)
        const end = Date.now()

        logInfos(data, start, params, options, time)

        if (options.overwriteData){
             this.#updateSimpleData(data)
        }
        return this
    }
```
and
```
function logInfos(data, start, parameters, options){
    options.logs && log("\n" + func.name + "()")
    options.logOptions && log("options:")
    options.logOptions && log(options)
    options.logParameters && log("parameters:")
    options.logParameters && log(parameters.slice(0, parameters.length - 1))
     options.logs | options.showDataNoOverwrite && showTable(data, options)
    options.logs && log(`Done in ${((end - start) / 1000).toFixed(3)} sec.`)
}
```